extensions/wikia/SpecialHealthcheck/HealthController.class.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+/**
+ * Extra health checks controller.
+ *
+ * Currently supports requests for getting the status of database clusters
+ */
+class HealthController extends WikiaController {
+
+	private $current = '';
+
+	private $clusters = [ ];
+	private $errors = [ ];
+	private $messages = [ ];
+	private $status = [ ];
+
+	/**
+	 * Get status of database clusters
+	 *
+	 * Returns:
+	 * - clusters: associative array of cluster statuses
+	 *   cluster name is the key and one of "ok", "warning", "critical" as a value
+	 * - messages: list of notices
+	 * - errors: list of errors
+	 */
+	public function databases() {
+		$cluster = $this->getVal( 'cluster' );
+		$clusters = $this->getAllClusters();
+		if ( $cluster ) {
+			$clusters = array_intersect( [ $cluster ], $clusters );
+		}
+
+		$this->clusters = $clusters;
+
+		$this->testClusters();
+
+		$this->setVal( 'clusters', $this->status );
+		if ( $this->errors ) {
+			$this->setVal( 'errors', $this->errors );
+		}
+		if ( $this->messages ) {
+			$this->setVal( 'messages', $this->messages );
+		}
+
+		$this->setVal('readWrite',[
+			'status' => !wfReadOnly(),
```

Comment:
``wfReadOnly`` gets data on-per cluster basis. It will return true only when read-only mode is set globally or on cluster A (where `SpecialHealthcheck` is enabled).